dtls_debug.h

```diff
@@ -73,6 +73,23 @@ log_t dtls_get_log_level(void);
 /** Sets the log level to the specified value. */
 void dtls_set_log_level(log_t level);
 
+/**
+ * Logging callback handler definition.
+ *
+ * @param level One of the DTLS_LOG_* values.
+ * @param message Zero-terminated string message to log.
+ */
+typedef void (*dtls_log_handler_t) (log_t level, const char *message);
+
+/* So applications using TinyDTLS know that this handler is available */
+#define HAVE_DTLS_SET_LOG_HANDLER
```

Comment:
Hm, wouldn't applications' configure just check for the existence of `dtls_set_log_handler()` in the library and then set whatever symbol they want? I am not sure that we want to maintain this preprocessor symbol all the time.

Reply:
It could, but it cannot test for the presence if the library is yet to be built as, say for a submodule. Another way would be to check against the version number of the TinyDTLS library so I went for the easy option.

Reply:
Removed the need for HAVE_DTLS_SET_LOG_HANDLER to be defined and code pushed.